src/coreclr/utilcode/util.cpp

```diff
@@ -1039,22 +1039,21 @@ DWORD LCM(DWORD u, DWORD v)
     // Vast majority of time, CPUGroupInfo is initialized in case 1. or 2.
     // The chance of contention will be extremely small, so the following code should be fine
     //
-retry:
     if (IsInitialized())
         return;
 
     if (InterlockedCompareExchange(&m_initialization, 1, 0) == 0)
     {
         InitCPUGroupInfo();
-        m_initialization = -1;
+        VolatileStore(&m_initialization, -1L);
```

Comment:
Reading/writing `m_initialization` without any barriers seemed like a potential race condition, so I am changing the code to use `VolatileLoad`/`VolatileStore`.  There should be no performance-critical code path calling this function.